integ/components/deadline/deadline_05_secretsManagement/test/deadline_05_secretsManagement.test.ts

```diff
@@ -0,0 +1,111 @@
+  }
+});
+
+describe('Deadline Secrets Management tests', () => {
+  test(`SM-${testSuiteId}-1: Deadline Repository configures Secrets Management`, async () => {
+    /**********************************************************************************************************
+     * TestID:          SM-1
+     * Description:     Confirm that Deadline Repository configures Secrets Management
+     * Input:           Output from "deadlinecommand secrets ListAllAdminUsers" call delivered via SSM command
+     * Expected result: List of admin users which shows that the user created by RFDK is registered
+    **********************************************************************************************************/
+    // GIVEN
+    const params = {
+      DocumentName: 'AWS-RunShellScript',
+      Comment: 'Execute ListAllAdminUsers via test script SM-run-secrets-command.sh',
+      InstanceIds: [bastionId],
+      Parameters: {
+        commands: [
+          `sudo -u ec2-user ~ec2-user/testScripts/SM-run-secrets-command.sh '${AWS.config.region}' '${smSecretArns[testSuiteId]}' ListAllAdminUsers`,
+        ],
+      },
+    };
+    const secret = await secretsManager.getSecretValue({ SecretId: smSecretArns[testSuiteId] }).promise();
+    const smCreds = JSON.parse(secret.SecretString!);
+    const adminUserRegex = new RegExp(`${smCreds.username}\\s+Registered`);
+
+    // WHEN
+    const response = await awaitSsmCommand(bastionId, params);
+
+    // THEN
+    expect(response.output).toMatch(adminUserRegex);
+  });
+
+  test(`SM-${testSuiteId}-2: Deadline Render Queue has a Server role`, async () => {
+    /**********************************************************************************************************
+     * TestID:          SM-2
+     * Description:     Confirm that Deadline Render Queue configures itself as a Server role
+     * Input:           Output from "deadlinecommand secrets ListAllMachines" call delivered via SSM command
+     * Expected result: List of machine identities contains an entry for the Render Queue machine and it has
+     *                  the Server role assigned to it
+    **********************************************************************************************************/
+    // GIVEN
+    const params = {
+      DocumentName: 'AWS-RunShellScript',
+      Comment: 'Execute ListAllMachines via test script SM-run-secrets-command.sh',
+      InstanceIds: [bastionId],
+      Parameters: {
+        commands: [
+          `sudo -u ec2-user ~ec2-user/testScripts/SM-run-secrets-command.sh '${AWS.config.region}' '${smSecretArns[testSuiteId]}' ListAllMachines "*" Server`,
+        ],
+      },
+    };
+
+    // WHEN
+    const response = await awaitSsmCommand(bastionId, params);
+
+    // THEN
+    // Assert there is exactly one Server identity (the RCS)
+    expect(response.output.match(/ec2-user\s+\[Server\]\s+Registered/)?.length).toBe(1);
```

Comment:
nit: this might help improve the output when this test fails so that we can see the actual deadline command output in the test failure message:

```suggestion
    expect(response.output).toMatch(/ec2-user\s+\[Server\]\s+Registered/);
```

I'm okay deferring this for later since the test will still fail on error and developers can always go to the SSM Run Command console in the account to obtain the actual output.

Reply:
As discussed offline, I've created a custom jest matcher using `expect.extends` to achieve what we want here with a more helpful error message if the test fails. Let me know what you think

Reply:
Wow - great job! bonus points 🥇 💯